In [1]:
import pandas as pd

# Load the UCI Student Performance mathematics dataset
# The CSV file uses semicolon ; as the separator

df = pd.read_csv("student-mat.csv", sep=";")

print("Dataset shape:", df.shape)

df.head()

Dataset shape: (395, 33)


,school,sex,age,address,famsize,Pstatus,Medu,Fedu,Mjob,Fjob,...,famrel,freetime,goout,Dalc,Walc,health,absences,G1,G2,G3
0,GP,F,18,U,GT3,A,4,4,at_home,teacher,...,4,3,4,1,1,3,6,5,6,6
1,GP,F,17,U,GT3,T,1,1,at_home,other,...,5,3,3,1,1,3,4,5,5,6
2,GP,F,15,U,LE3,T,1,1,at_home,other,...,4,3,2,2,3,3,10,7,8,10
3,GP,F,15,U,GT3,T,4,2,health,services,...,3,2,2,1,1,5,2,15,14,15
4,GP,F,16,U,GT3,T,3,3,other,other,...,4,3,2,1,2,5,4,6,10,10


# CSV Loading Explanation
Summary

The CSV file appears to have loaded correctly because the dataset shape is (395, 33), indicating that the file contains 395 student records and 33 variables. The output from df.head() also displays the first five rows of the dataset, confirming that the data was successfully loaded into a pandas DataFrame.

Interpretation

Two useful checks for confirming that the CSV loaded correctly are df.shape and df.head(). The first check verifies the number of rows and columns in the dataset. The second check displays the first few rows so that the column names and values can be visually inspected. The separator is important because the UCI student-mat.csv file uses semicolons rather than commas. If the wrong separator is used, pandas may load the entire row into a single column instead of splitting the dataset into 33 columns.

Recommendation

The next step is to inspect the dataset more carefully by examining the column names, data types, and missing values. This can be done using:

print(df.columns)

print(df.info())

print(df.isnull().sum())

These checks help verify data quality before performing data analysis or machine learning.


# Verification Statement
I verified that student-mat.csv loaded correctly because the DataFrame shape is (395, 33), matching the UCI documentation: 395 student records and 33 variables.


# Reflection Question
If the DataFrame were loaded as a single column, I would first check the separator used in pd.read_csv(). The UCI student-mat.csv file is semicolon-separated, so I need to use sep=";". If the wrong separator is used, pandas may read each full row as one column instead of splitting the data into 33 columns.
